libs/cardano-ledger-core/src/Cardano/Ledger/EpochBoundary.hs

```diff
@@ -181,15 +186,19 @@ instance CC.Crypto c => FromSharedCBOR (SnapShot c) where
     pure SnapShot {ssStake, ssDelegations, ssPoolParams}
 
 -- | Snapshots of the stake distribution.
+--
+-- Note that ssStakeMark and ssStakeMarkPoolDistr are lazy on
+-- purpose since we only want to force the thunk after one stability window
+-- when we know that they are stable (so that we do not compute them if we do not have to).
```

Comment:
Maybe adding a url to the ADR in this PR:
```suggestion
-- when we know that they are stable (so that we do not compute them if we do not have to).
-- See more info in the [Optimize TICKF ADR](https://github.com/input-output-hk/cardano-ledger/blob/master/docs/adr/2022-12-12_007-optimize-ledger-view.md)
```